src/controller/stream-controller.ts

```diff
@@ -769,7 +769,37 @@ export default class StreamController
               3,
             )}`,
           );
-          media.currentTime = liveSyncPosition;
+
+          if (this.config.liveSyncMode === 'buffered') {
+            const bufferInfo = BufferHelper.bufferInfo(
+              this.media,
+              liveSyncPosition,
+              0,
+            );
+
+            if (!bufferInfo || bufferInfo.buffered?.length === 0) {
+              media.currentTime = liveSyncPosition;
+              return;
+            }
+
+            const isLiveSyncInBuffer = bufferInfo.start <= currentTime;
+
+            if (isLiveSyncInBuffer) {
+              media.currentTime = liveSyncPosition;
+              return;
+            }
+
+            const { nextStart } = BufferHelper.bufferInfo(
+              media,
+              currentTime,
+              0,
+            );
```

Comment:
Use `bufferedInfo` here since we already have `buffered` ranges to operate on (saves a step in `BufferHelper`).

```suggestion
            const { nextStart } = BufferHelper.bufferedInfo(
              bufferInfo.buffered,
              currentTime,
              0,
            );
```